### PR TITLE
Fix flaky capture service start

### PR DESCRIPTION
**Pull request by zbud-msft**
#### Why I did it

Dropping control character (message sent when XSUB connects to XPUB as part of ZMQ Proxy setup to notify that subscription has been made) in do capture has been flaky since control character is not guaranteed to be the first message sent if there are events (like event-down-ctr) being published to XSUB.

Scenarios

1) Control character is sent and is first message when starting capture service

`eventd#eventd#eventd: :- heartbeat_ctrl: Set heartbeat_ctrl pause=1`
`eventd#eventd#eventd: :- do_capture: Received subscription message when XSUB connects to XPUB`

2) Events like event-down ctr is sent before control character

`eventd#eventd#eventd: :- run: Dropping Message: 22 serialization::archive 18 17 sonic-events-host`
`eventd#eventd#eventd: :- run: Dropping Message: 22 serialization::archive 18 0 0 4 0 0 0 1 d 103 {"sonic-events-host:event-stopped-ctr":{"ctr_name":"EVENTD","timestamp":"2024-08-27T00:02:51.407518Z"}} 1 r 36 3357542f-bae1-458f-a804-660e620d21f5 1 s 1 9 1 t 19 1724716971407591080`
`heartbeat_ctrl: Set heartbeat_ctrl pause=1`
`do_capture: Received subscription message when XSUB connects to XPUB`

3) Control character is not sent at all

`eventd#eventd#eventd: :- heartbeat_ctrl: Set heartbeat_ctrl pause=1`

4) Control character is delayed and not caught when starting capture service, but is then caught after causing deserialize error.

`do_capture: Receiving event from source: 22 serialization::archive 18 17 sonic-events-host, will read second part of event`
`deserialize: deserialize Failed: input stream errorstr[0:64]:(#001) data type: std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&`
`zmq_read_part: Failed to deserialize part rc=-2`
`zmq_read_part: last:errno=11`
`zmq_message_read: Failure to read part1 rc=-2`
`zmq_message_read: last:errno=11`

We can cover these scenarios by just dropping the control character inside zmq_message_read as part of events_common in swsscommon (different PR). In this PR we will remove such handling logic and make sure that empty events that will be sent by control character are ignored.

##### Work item tracking
- Microsoft ADO **(number only)**:28728116

#### How I did it

Remove logic for handling control character

#### How to verify it

UT and sonic-mgmt test cases.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

File: src/sonic-eventd/src/eventd.cpp [hunk 1]
```diff
@@ -409,8 +409,30 @@ capture_service::do_capture()
...
+
+        if(rc == 1) { // Expected case to receive subscription message as very first message
+            SWSS_LOG_INFO("Received subscription message when XSUB connects to XPUB");
+        } else if (rc > 1) { // If there are events already published to XSUB before XSUB connects to XPUB, we can receive events before subscription message
+            string event_source((const char*)zmq_msg_data(&msg), zmq_msg_size(&msg));
+            SWSS_LOG_DEBUG("Receiving event from source: %s, will read second part of event", event_source.c_str());
+            int more = 0;
+            size_t more_size = sizeof(more);
+            zmq_getsockopt(cap_sub_sock, ZMQ_RCVMORE, &more, &more_size);
+            if(more) {
```
Comment (liuh-80) [inline src/sonic-eventd/src/eventd.cpp:425]: Do we need a loop here to read all message?

Also, if there is an ADO please update the PR description.

File: src/sonic-eventd/src/eventd.cpp [hunk 2]
```diff
@@ -409,8 +409,30 @@ capture_service::do_capture()
...
          * This behavior will only happen once when XSUB connects to XPUB not everytime cache is started.
          *
+         * There are chances that there are events already published to XSUB endpoint before XSUB is able to connect to XPUB, so we can receive events
+           before the subscription message
          */
-         init_done = true;
+
+
+        if(rc == 1) { // Expected case to receive subscription message as very first message
+            SWSS_LOG_INFO("Received subscription message when XSUB connects to XPUB");
```
Comment (liuh-80) [inline src/sonic-eventd/src/eventd.cpp:418]: If there are too much "received subscription message" log, suggest change to SWSS_LOG_DEBUG.

Comment (zbud-msft) [reply src/sonic-eventd/src/eventd.cpp:418]: This will happen only once during process lifetime.

Comment (liuh-80) [reply src/sonic-eventd/src/eventd.cpp:418]: #Closed

File: src/sonic-eventd/src/eventd.cpp (hunk 1 above)
Comment (zbud-msft) [reply src/sonic-eventd/src/eventd.cpp]: Changed logic to not account for capture service capturing control character. Will be done in zmq_read_part

https://github.com/sonic-net/sonic-swss-common/pull/906

**Comment by zbud-msft** [comment]
Relies on https://github.com/sonic-net/sonic-swss-common/pull/906

**Review by liuh-80 — APPROVED** [review]

**Review by qiluo-msft — APPROVED** [review]

**Comment by zbud-msft** [comment]
@StormLiangMS Can we backport this to 202311?

**Pull request merged**